prospector/commit_preprocessor/test_preprocessor.py

```diff
@@ -41,8 +52,77 @@ def test_proprocess_commit_set(repository):
     assert len(preprocessed_commits) == len(commit_set)
 
 
+def test_extract_ghissue_references():
+    result = extract_ghissue_references(
+        "#1 blah #2 blah a#3 blah #4b blah c#5d blah 0#6 blah #10 44 #999 0 #6"
+    )
+    assert result == ["#1", "#2", "#3", "#4", "#5", "#6", "#10", "#999", "#6"]
```

Comment:
Not sure the assert is right: i guess a space is necessary before each hash sign....

Reply:
The current implementation of extract_ghissue_references that you provided extracts GH issues even if there is no white-space character before them. That's why we created the test like this, but indeed a space is required before th GH issue ID. We can change the test but the implementation (regexp) needs to be changed in the extract_ghissue_references as well so that it won't extract issue IDs without spaces in front of them (which it does in the current version). @idarav @Szamos96 will handle this.

Reply:
Right, please feel free to fix both (functionality + test).